ledger-core/inc/core/operation/OperationDatabaseHelper.inl

```diff
@@ -0,0 +1,184 @@
+/*
+ *
+ * OperationDatabaseHelper
+ * ledger-core
+ *
+ * Created by Alexis Le Provost on 01/10/2019.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <type_traits>
+
+#include <core/api/Amount.hpp>
+#include <core/api/BigInt.hpp>
+#include <core/bytes/Serialization.hpp>
+#include <core/crypto/SHA256.hpp>
+#include <core/database/SociNumber.hpp>
+#include <core/database/SociDate.hpp>
+#include <core/database/SociOption.hpp>
+#include <core/collections/Strings.hpp>
+#include <core/operation/Operation.hpp>
+#include <core/wallet/BlockDatabaseHelper.hpp>
+#include <core/wallet/TrustIndicator.hpp>
+
+namespace ledger {
+namespace core {
+
+namespace impl {
+
+    template <typename T, typename = void>
```

Comment:
Nice. Please add some comment about that trait so that people don’t have to read the whole series of declarations to get what it is.